src/protocol/mod.rs

```diff
@@ -292,6 +292,7 @@ impl ProxyHeader {
             buf.advance(1);
             return Err(Version1ParsingError::ExpectedCRLF.into());
         }
+        buf.advance(1);
```

Comment:
This looks sane, however perhaps it'd be better to just `get_u8` the byte and instead compare that? That way we advance 1 no matter what (which we do anyways), and avoid an unnecessary slice.

Reply:
Done, great suggestion